packages/firestore/src/local/indexeddb_persistence.ts

```diff
@@ -306,10 +305,13 @@ export class IndexedDbPersistence implements Persistence {
 
         this.scheduleClientMetadataAndPrimaryLeaseRefreshes();
 
-        return this.simpleDb.runTransaction(
+        return this.runTransaction(
+          'getHighestListenSequenceNumber',
           'readonly',
-          [DbTargetGlobal.store],
-          txn => getHighestListenSequenceNumber(txn)
+          txn =>
+            getHighestListenSequenceNumber(
```

Comment:
There's a `getHighestSequenceNumber` that does this translation for you, though I'm really dubious that we should really have names that differ in this way for the difference in transaction types.

It seems like there's a change warranted where we minimize the use of SimpleDbTransaction?

Reply:
The original version of this PR changed `getHighestListenSequenceNumber` to take an IndedexDbTransaction. I had to duplicate a couple lines of code in the SimpleDb schema conversion to make this work, since the schema conversion needs the highest sequence number but doesn't use IndedexDbTransactions. 

The other callsite that does this conversion is part the TargetCache, which hasn't been started at this point.

I have no preference either way. The cleanest solution was probably the one that changed all sites to use IndedexDbTransaction and added three lines of duplicate code to the schema conversion.

Reply:
It's a shame that we're ending up with two different transaction mechanisms and that migrations need to be written differently than the rest of the code. Is that the end state?

Is your intent to make everything mostly use IndedexDbTransaction except for the few places that can't? If so I'm not as concerned about the individual steps along the way.

Reply:
I cleaned this up a little, and hope it is an improvement.
- I inlined two of the helper methods and changed them to only use IndedexDbTransactions/PersitenceTransactions.
- I changed the called in IndexedDbPersistence to use the method from the TargetCache. The TargetCache doesn't have a `start()`, so there is actually no issue here.
- The SimpleDbSchema class now has a couple lines of code duplication.